protocol/amqp/v2/protocol.go

```diff
@@ -35,54 +30,60 @@ type Protocol struct {
 }
 
 // NewProtocolFromClient creates a new amqp transport.
-func NewProtocolFromClient(client *amqp.Client, session *amqp.Session, queue string, opts ...Option) (*Protocol, error) {
+func NewProtocolFromClient(
```

Comment:
question: are we ok with this breaking change on this signature? I understand why we technically might not need options anymore (since those are now structs), but you might want to keep the options pattern to not break this signature and allow for future expansions. You could introduce a new option to set sender/receiver options on the protocol.

Reply:
Or are you worried that those two structs are always needed and thus an options pattern doesn't really apply?